ats/ast/nodes.py

```diff
@@ -90,9 +90,11 @@ def __init__(self, name: str, stmt_lst_node: StmtLstNode):
 
 
 class ProgramNode(ASTNode):
-    def __init__(self, procedure_node: ProcedureNode):
+    def __init__(self, procedure_nodes):
         super().__init__()
-        self.add_node("procedure", procedure_node)
+
+        self.add_node("procedure", procedure_nodes[0])
+        self.add_node("procedures", procedure_nodes[1:])
```

Comment:
Ogolnie bardzo mi sie podoba pomysl, ale sama nazwa `procedures` sugeruje raczej wszystkie procedury 

```suggestion
        self.add_node("procedures", procedure_nodes)
```